src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -21,55 +44,748 @@ const ActorInit En_Insect_InitVars = {
     (ActorFunc)EnInsect_Update,
     (ActorFunc)EnInsect_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BE20.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Insect/func_80A7BE40.s")
+static ColliderJntSphItemInit sColliderItemInit[1] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 0, { { 0, 0, 0 }, 5 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sColliderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x19, 0x10, COLSHAPE_JNTSPH },
+    1,
+    sColliderItemInit,
+};
+
+u16 D_80A7DF10[] = { 0, 5, 7, 7 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 700, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 20, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 600, ICHAIN_STOP),
+};
+
+Vec3f D_80A7DF28[2] = { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } };
+
+void func_80A7BE20(EnInsect* this) {
+    this->unk_314 = D_80A7DF10[this->actor.params & 3];
+}
+
+f32 EnInsect_XZDistanceSquared(Vec3f* v1, Vec3f* v2) {
+    return SQ(v1->x - v2->x) + SQ(v1->z - v2->z);
+}
+
+s32 func_80A7BE6C(EnInsect* this, GlobalContext* globalCtx) {
+    u32 pad;
+    Player* player = PLAYER;
+    Vec3f pos;
+
+    if (this->actor.xzDistFromLink < 32.0f) {
+        pos.x = Math_Sins(this->actor.yawTowardsLink + 0x8000) * 16.0f + player->actor.posRot.pos.x;
+        pos.y = player->actor.posRot.pos.y;
+        pos.z = Math_Coss(this->actor.yawTowardsLink + 0x8000) * 16.0f + player->actor.posRot.pos.z;
+
+        if (EnInsect_XZDistanceSquared(&pos, &this->actor.posRot.pos) <= 400.0f) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+void func_80A7BF58(EnInsect* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_040341FC, 1.0f, 0.0f, 0.0f, 1, 0.0f);
+}
+
+/**
+ * Find the nearest soft dirt patch within 6400 units in the xz plane and the current room
+ * Returns 1 if one was found, 0 otherwise
+ */
```

Comment:
I believe theres certain doxygen syntax to describe returns. I dont know it cause I havent really used it yet, but theres a guide for using doxygen in decs